src/app/core/import/edit-import-column-mapping/edit-import-column-mapping.component.ts

```diff
@@ -0,0 +1,94 @@
+import { Component, Input, Output, EventEmitter, inject } from "@angular/core";
+import { ColumnMapping } from "../column-mapping";
+import { EntityConstructor } from "../../entity/model/entity";
+import { FormFieldConfig } from "../../common-components/entity-form/FormConfig";
+import { DefaultDatatype } from "../../entity/default-datatype/default.datatype";
+import { MatDialog } from "@angular/material/dialog";
+import { ComponentRegistry } from "../../../dynamic-components";
+import { EntitySchemaService } from "../../entity/schema/entity-schema.service";
+import { HelpButtonComponent } from "app/core/common-components/help-button/help-button.component";
+import { MatInputModule } from "@angular/material/input";
+import { EntityFieldSelectComponent } from "app/core/entity/entity-field-select/entity-field-select.component";
+import { FormsModule } from "@angular/forms";
+import { MatButtonModule } from "@angular/material/button";
+import { MatBadgeModule } from "@angular/material/badge";
+import { MappingDialogData } from "../import-column-mapping/import-column-mapping.component";
+
+@Component({
+  selector: "app-edit-import-column-mapping",
+  templateUrl: "./edit-import-column-mapping.component.html",
+  styleUrls: ["./edit-import-column-mapping.component.scss"],
+  standalone: true,
+  imports: [
+    HelpButtonComponent,
+    MatInputModule,
+    EntityFieldSelectComponent,
+    FormsModule,
+    MatButtonModule,
+    MatBadgeModule,
+  ],
+})
+export class EditImportColumnMappingComponent {
+  private dialog = inject(MatDialog);
+  private componentRegistry = inject(ComponentRegistry);
+  private schemaService = inject(EntitySchemaService);
+
+  @Input() col: ColumnMapping;
+  @Input() entityCtor: EntityConstructor;
+  @Input() usedColNames: Set<string>;
+  @Input() rawData: any[];
+  @Input() columnMapping: ColumnMapping[] = [];
+
+  @Output() valueChange = new EventEmitter<ColumnMapping[]>();
```

Comment:
Let the Output also only relate to that mapping that this component is responsible for, i.e. 

```suggestion
  @Output() valueChange = new EventEmitter<ColumnMapping>();
```

and then `this.valueChange.emit(this.col)`, I think.

Reply:
As far as I can see, we can then also completely remove the `@Input() columnMapping: ColumnMapping[]` for the component here.